src/models/signature-chain-link.ts

```diff
@@ -0,0 +1,21 @@
+import { prop } from "@typegoose/typegoose";
+import Form from "./form";
+import { SigningPositions } from "@/enums/signing-positions";
+import Position from "./position";
+
+class SignatureChainLink {
+
+    @prop({ required: true, ref: () => Form, type: () => Form })
+    public form!: Form;
+
+    @prop({ required: true, type: () => String, enum: SigningPositions })
+    public position!: SigningPositions;
+
+    @prop({ required: true, ref: () => Position, type: () => Position })
+    public specificPosition!: Position; //! Only thing I'm confused about is why a SignatureChainLink has a specific position if its supposed to be generic?
```

Comment:
So when we create a list of people who need to sign a form, it can be something generic like "manager", "department head", etc. where the exact person who it's going to be depends on who is initiating the form. It could also be a very specific position that needs to sign it. For example, there are a lot of forms where the "HR director" needs to sign it, and that's going to be the same position every time

Reply:
Oh I see, I've changed SignatureChainLink --> GenericSignee and now it just has a position and specificPosition (which is now optional) so that a form has an array of GenericSignees